swap/src/cli/cancel_and_refund.rs

```diff
@@ -105,11 +112,32 @@ pub async fn refund(
     };
 
     tracing::info!(%swap_id, "Manually refunding swap");
-    state6.publish_refund_btc(bitcoin_wallet.as_ref()).await?;
 
-    let state = BobState::BtcRefunded(state6);
-    db.insert_latest_state(swap_id, state.clone().into())
-        .await?;
+    match state6.publish_refund_btc(bitcoin_wallet.as_ref()).await {
+        Ok(_) => {
+            let state = BobState::BtcRefunded(state6);
+            db.insert_latest_state(swap_id, state.clone().into())
+                .await?;
 
-    Ok(state)
+            Ok(state)
+        }
+        Err(error) => {
+            if state6
+                .check_for_tx_punish(bitcoin_wallet.as_ref())
```

Comment:
I think here we should check for the timelock instead of for the punish transaction.

The punish timelock might have expired which means we cannot refund anymore and have (from Bobs perspective) already been punished. But Alice might not have published the punish transaction yet which means we won't find it on the Blockchain. This'll also revert the breaking database change.